### ISSUE TITLE
OSX Silicon Support

### ISSUE THREAD

**Issue by eoliphan** [uncomplicate/neanderthal#125]
Would be nice to have "native" support on next-gen macs.  Looks like the [Accelerate](https://developer.apple.com/documentation/accelerate/veclib) library provides BLAS and LAPACK

**Comment by blueberry** [comment]
Yes.I intend to do something about this eventually. I'll see what I can do once I get hold of the appropriate hardware. However, I use lots of native functions that are not available in BLAS and LAPACK. That means that i won't be able to support full functionality.

**Comment by kxygk** [comment]
It'd be nice if there was some generic (either pure Clojure or JVM) fallback. The current backend is obviously as close to optimal as possible in terms of speed - but it's a pain to distribute/bundle.  I often work with non-huge matrices and I don't really need to squeeze everything I can out of every CPU cycle - and I end up having to fall back onto `core.matrix` ... which is really kinda buggy and not as polished as `neanderthal`

I get this might not be on your radar though since I think you're using this for ML :)

**Comment by mbjarland** [comment]
@blueberry Would it be helpful if you had access to an m1 instance in the cloud?  If yes I could have my company sponsor the instanse.

**Comment by blueberry** [comment]
It's difficult to say. If it was just a matter of recompiling, then yes. But for OSX support, I would have to develop an independent engine from scratch, with Apple's proprietary BLAS/LAPACK. I'd have to familiarize myself with their ecosystem to be able to do that. Thank you for the offer, though. This might be an option if I don't find a better solution.

**Comment by blueberry** [comment]
Just a quick update that this is brewing.

**Comment by blueberry** [comment]
The initial support for Apple M is now (as of 0.53.0) available for the core and linalg namespaces. Other namespaces are on their way, so I'll close this for now.
